dask_cuda/dask_cuda_worker.py

```diff
@@ -252,6 +264,14 @@ def del_pid_file():
             contact_address=None,
             env={"CUDA_VISIBLE_DEVICES": cuda_visible_devices(i)},
             name=name if nprocs == 1 or not name else name + "-" + str(i),
+            data=DeviceHostFile(
+                device_memory_limit=get_device_total_memory(index=i)
+                if (device_memory_limit == "auto" or device_memory_limit == int(0))
+                else parse_bytes(device_memory_limit),
+                memory_limit=parse_memory_limit(
+                    memory_limit, nthreads, total_cores=nprocs
+                ),
+            ),
```

Comment:
I recommend the following instead:

```python
data=(DeviceHostFile, {...})
```

Otherwise we create the data object immediately here, and then need to pass it down to the worker through a process boundary.

This is implemented here, but it looks like it's not documented anywhere (my mistake)

https://github.com/dask/distributed/blob/8e449d392e91eff0a3454ee98ef362de8f78cc4f/distributed/worker.py#L500-L501

Also, if the user specifies `device_memory_limit=0` then we might want something simpler.  I can imagine wanting to turn off this behavior if things get complex.

We probably also want the same treatment in `LocalCUDACluster`, though as a first pass we can also include this ourselves.

Reply:
I was in the process of doing that, only saw your comment now. Indeed, this is a better solution, but either way, the downside is that I had to create the work directory in `dask-cuda-worker`, which I'm not particularly happy with, so if you have a suggestion on how to avoid that, it would be great!

I'm already working on the `LocalCUDACluster`, I'm still not done, will soon push those.

Reply:
> the downside is that I had to create the work directory in dask-cuda-worker, which I'm not particularly happy with, so if you have a suggestion on how to avoid that, it would be great!

Hrm, yes I can see how that would be a concern.  I don't have a good solution currently unfortunately.  I'll think about it though.